resource-management/pkg/common-lib/types/runtime/interfaces.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2022 Authors of Global Resource Service.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"global-resource-service/resource-management/pkg/common-lib/types"
+	"global-resource-service/resource-management/pkg/common-lib/types/location"
+	"time"
+)
+
+type Object interface {
```

Comment:
please add comments here for readability. e.g. instead of a generic interface{} in go lang ? and the components that uses the runtime object can cast ?

Reply:
> please add comments here for readability. e.g. instead of a generic interface{} in go lang ? and the components that uses the runtime object can cast ? if the effort is too big for 930, let's add a TODO to track future improvement.

Comment added. I don't think we need to add all types that runtime objects can be casted to. It depends on interface implementation.